### PR TITLE
Kafka external liveness test

### PR DESCRIPTION
**Pull request by mtarng**
Building off the last PR #11 for https://github.com/microsoft/bedrock/issues/520

This external liveness test requires a few things mentioned in the README. I've also added the loadbalancer without tls support.

@sayar Do we want to change the static manifests into a helm chart so we can modify the manifests as needed? 

This PR is more of a checkup on where I'm heading. Looking into adding TLS/SSL support now.

### CODE REVIEW

**Comment by mtarng** [comment]
Also aware that there's a bunch of duplicate code between the external and internal tests, will need to refactor.

**Review by andrebriggs — COMMENTED** [review]
I ran the internal test and got the following:

<pre>
$ sh internaltest.sh 
secret "mirror-maker-cluster-ca-cert" created
kafkatopic.kafka.strimzi.io "test" created
kafkatopic.kafka.strimzi.io "test-one-rep" created
kafkatopic.kafka.strimzi.io "test-replication" created
kafkauser.kafka.strimzi.io "test-user" created
statefulset.apps "kafkaclient" created
2f90163d-286a-4832-8bde-1a0e6a5aef0e
Test Topic: topic-2f90163d-286a-4832-8bde-1a0e6a5aef0e
OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
Created topic topic-2f90163d-286a-4832-8bde-1a0e6a5aef0e.
Creating Input Message file.
E2B1A6EF-91F0-499C-8943-4118A636589A
BEB6180F-2BCB-4A1E-B52F-8368A056D314
B4B24EED-399F-4EB3-B292-3CA73A91E8BD
41EF2404-6E27-4DB1-AA13-115157CC9EAF
73FD16E5-01DF-4BD6-ACDF-4760D46554C7
D6DBDAC5-E6F1-4566-9B65-7E81F77303D7
51EA8106-6546-43DD-A8A2-3C3FA66979DE
9509ACF8-7236-478F-8866-9C0AD80CA854
549360BE-2688-4F5A-A3DD-5CF32DFD66E1
193C41B1-3275-4211-8AB8-F727E5C47FDA
<b>error: unable to upgrade connection: container not found ("kafka")
error: unable to upgrade connection: container not found ("kafka")
internaltest.sh: line 51: kill: (53243) - No such process</b>
listing topics
OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
test
test-one-rep
test-replication
topic-2f90163d-286a-4832-8bde-1a0e6a5aef0e
kafkatopic.kafka.strimzi.io "test" deleted
kafkatopic.kafka.strimzi.io "test-one-rep" deleted
kafkatopic.kafka.strimzi.io "test-replication" deleted
kafkauser.kafka.strimzi.io "test-user" deleted
statefulset.apps "kafkaclient" deleted
secret "mirror-maker-cluster-ca-cert" deleted
deleting test topic
kubectl exec -n kafka -ti kcluster-kafka-0 --container kafka -- bin/kafka-topics.sh --zookeeper localhost:2181 --delete --topic topic-2f90163d-286a-4832-8bde-1a0e6a5aef0e
OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
Topic topic-2f90163d-286a-4832-8bde-1a0e6a5aef0e is marked for deletion.
Note: This will have no impact if delete.topic.enable is not set to true.
listing topics after deletion
OpenJDK 64-Bit Server VM warning: If the number of processors is expected to increase from one, then you should configure the number of parallel GC threads appropriately using -XX:ParallelGCThreads=N
<b>Test Failed!!! - There's a difference between input and output!!!</b>
</pre>

This was because my Kafka broker pods weren't ready. We may let the user know that their Kafka broker pods aren't completely up yet. This could effect the external test as well

File: test/README.md [hunk 1]
```diff
@@ -5,11 +5,27 @@ This directory contains simple test scripts to check if a Strimzi Operator Kafka
 ## Requirements
 
 - A kubernetes cluster with Kafka deployed via Strimzi Operator
+
+## Liveness Check from Inside the cluster
+### Test Requirements
 - `kubectl` should be configured to point to this cluster
 
-## Liveness Check
 ### Running the test
```
Comment (andrebriggs) [inline test/README.md:13]: @mtarng I think we should call out that these tests only going through one (hardcoded) broker. Perhaps we can have a `TODO` section on this readme to indicate things we want to improve in the future

File: test/README.md [hunk 2]
```diff
@@ -5,11 +5,27 @@ This directory contains simple test scripts to check if a Strimzi Operator Kafka
...
+
+The test results will displayed at the end. The script with exit with code 0 for success and 1 for failure.
+
+## Liveness Check from Outside the cluster
+### Test Requirements
+- `kubectl` should be configured to point to this cluster
+- The Kafka deployment should be configured with an external loadbalancer listener.
+  - [Reference Guide](https://strimzi.io/2019/05/13/accessing-kafka-part-4.html)
+  - Currently TLS is _not_ supported for this test
+- `kafkacat` needs to be installed and added to $PATH
```
Comment (andrebriggs) [inline test/README.md:25]: Perhaps mention `brew install kafkacat`?

**Pull request merged**